src/reflect/value.go

```diff
@@ -1395,7 +1395,24 @@ func (v Value) SetMapIndex(key, elem Value) {
 
 // FieldByIndex returns the nested field corresponding to index.
 func (v Value) FieldByIndex(index []int) Value {
-	panic("unimplemented: (reflect.Value).FieldByIndex()")
+	if len(index) == 1 {
+		return v.Field(index[0])
+	}
+	if v.Kind() != Struct {
+		panic(&ValueError{"FieldByIndex", v.Kind()})
+	}
+	for i, x := range index {
+		if i > 0 {
+			if v.Kind() == Pointer && v.typecode.elem().Kind() == Struct {
+				if v.IsNil() {
+					panic("reflect: indirection through nil pointer to embedded struct")
+				}
+				v = v.Elem()
+			}
```

Comment:
Interesting that this is needed for Value but not for Type. I guess the logic is that you'd want to index into fields like in plain Go (`.fieldName`) but that it doesn't make much sense for types?